scrapers_next/al/people.py

```diff
@@ -1,151 +1,113 @@
-            elif type == "Email:":
-                email = info
-
-        address = f"{street}, {office}, {city} AL"
-
-        image = (
-            CSS("#ContentPlaceHolder1_TabSenator_TabLeg_imgLEG")
-            .match_one(self.root)
-            .get("src")
-        )
-
-        p = ScrapePerson(
-            name=full_name.title(),
-            state="al",
-            chamber=self.input.chamber,
-            party=party,
-            district=district,
-            email=email,
-            image=image,
-        )
-        p.add_source(self.source.url)
-        p.add_source(self.input.url)
-
-        # This address is the capitol office
-        if "11 South Union Street" in street:
-            p.capitol_office.address = address
-            p.capitol_office.voice = phone
-            try:
-                p.capitol_office.fax = fax
-            except ValueError:
-                pass
-        else:
-            p.district_office.address = address
-            p.district_office.voice = phone
-            try:
-                p.district_office.fax = fax
-            except ValueError:
-                pass
-
-        p.extras["postal code"] = postal
-        p.extras["county"] = county
-
-        return p
-
-
-class SenList(HtmlListPage):
-    selector = XPath("//input[@type='image']")
-    source = URL(
-        "https://www.legislature.state.al.us/aliswww/ISD/Senate/ALSenators.aspx",
-        timeout=30,
+        data = self.response.json().get("data")
+
+        # Combine all data fields that contain a member object
+        members = []
+        member_fields = [
+            "houseMembersNewImgUrl",
+            "houseLeadersNewImgUrl",
+            "senateMembersNewImgUrl",
+            "senateLeadersNewImgUrl",
+        ]
+        for field in member_fields:
+            # Ignore fields that are missing
+            if data.get(field):
+                members += data.get(field)
+
+        for member in members:
+            # Skip adding member if district info isn't found.
+            if not member.get("District") or member.get("District") == "Not Available":
+                name = member.get("FullName")
+                self.logger.warning(f"{name} has no listed district, skipping.")
+                continue
+
+            # Remove extra text from district
+            district = member.get("District").split(" ")[-1]
+            print(district)
+
+            p = ScrapePerson(
+                name=member.get("FullName"),
+                state="al",
+                chamber=self.chamber,
+                party=member.get("Affiliation"),
+                district=district,
+                email=member.get("Email"),
+                image=member.get("NewImgUrl"),
+                given_name=member.get("FirstName"),
+                family_name=member.get("LastName"),
+            )
+
+            p.add_source(
+                self.source.url, note="graphql endpoint for member information"
+            )
+
```

Comment:
Let's also add a link for HTML so users can see the HTML pages. Since there are not individual URL paths for each member, it's fine to provide the path for the page with the tabs with leaders and members.

So if their chamber is lower, link is:
"https://alison.legislature.state.al.us/house-leaders-members"

And if upper:
"https://alison.legislature.state.al.us/senatee-leaders-members"

Similar to with committee scraper, let's just make the note="homepage"